### PR TITLE
Refactor page size for storage

### PR DESCRIPTION
**Pull request by agourlay**
Extract the hardcoded page size to enable testing different values.

I found a few bugs on the way :)

### CODE REVIEW

File: src/payload_storage.rs [hunk 1]
```diff
@@ -105,14 +111,14 @@ impl PayloadStorage {
         if self.pages.is_empty() {
             return None;
         }
-        let needed_size =
-            SlottedPageMmap::MIN_VALUE_SIZE_BYTES.max(payload_size) + SlotHeader::size_in_bytes();
```
Comment (agourlay) [inline src/payload_storage.rs:109]: fixed encapsulation issue, the storage layer should not know about the slot sizes

File: src/payload_storage.rs [hunk 2]
```diff
@@ -546,23 +564,23 @@ mod tests {
...
                 _ => unreachable!(),
             }
         }
     }
 
-    #[test]
-    fn test_behave_like_hashmap() {
-        let (dir, mut storage) = empty_storage();
+    #[rstest]
+    fn test_behave_like_hashmap(#[values(10, 10_000, 32_000_000)] page_size: usize) {
```
Comment (agourlay) [inline src/payload_storage.rs:582]: test different page sizes

File: src/slotted_page.rs [hunk 3]
```diff
@@ -251,16 +250,12 @@ impl SlottedPageMmap {
...
-            != 0
+        self.free_space() >= SlotHeader::size_in_bytes() + SlottedPageMmap::MIN_VALUE_SIZE_BYTES
     }
 
     /// Check if there is enough space for a new slot + value
     fn has_capacity_for_value(&self, extra_len: usize) -> bool {
-        self.free_space()
-            .saturating_sub(SlotHeader::size_in_bytes() + extra_len)
-            != 0
+        self.free_space() >= SlotHeader::size_in_bytes() + extra_len
```
Comment (agourlay) [inline src/slotted_page.rs:258]: off by one error, equality is a valid state

**Review by coszio — APPROVED** [review]
Very good cleanup and fixes, thanks!

**Pull request merged**